controllers/user.js

```diff
@@ -125,16 +138,36 @@ async function add(req, res) {
   res.redirect("/user/preferences");
 }
 
-async function bundleVerifiedEmails(email, ifPrimary, id, verificationStatus, allBreaches) {
+function getResolvedBreachesForEmail(user, email) {
+  if (typeof user.resolved_breaches !== "object") {
+    return [];
+  }
+  return user.resolved_breaches.hasOwnProperty(email) ? user.resolved_breaches[email] : [];
+}
+
+function annotateFoundBreachesAsResolvedOrNot(foundBreaches, resolvedBreaches) {
+  const annotatedFoundBreaches = [];
+  foundBreaches.forEach( (breach, index) => {
+    const IsResolved = resolvedBreaches.includes(index) ? true : false;
```

Comment:
Using an `index` feels odd, since I'm not 100% sure how reliable those are? Couldn't they change if new breaches are added/removed/resorted? Wasn't sure if we could use something like the `breach.Name` or something more reliable.

Also, I believe `.includes()` returns a Boolean, so the `? true : false` probably isn't needed.

Reply:
Well, I guess this is answered below in the excellent comment:

>         // NOTE: DO NOT CHANGE THIS SORT LOGIC
>         // We store breach resolutions by recency indices,
>         // so that our DB does not contain any part of any user's list of accounts